### ISSUE TITLE
Does the user need to restore engine state when resuming training from checkpoints?

### ISSUE THREAD

**Issue by szhang104** [pytorch/ignite#638]
I noticed that the provided `ModelCheckpoint` does not come with a corresponding hook for loading checkpoints to resume training, so normally people just manually use vanilla `torch.load()` to restore the `state_dict`s of model, optimizer and lr_scheduler.

Can any developer confirm if the user need to save and restore the training and evaluation engine states as well? Will restarting from a blank engine state affect the training results in any noticeable way?

**Comment by vfdev-5** [comment]
@szhang104 we are working on simple resuming training. 
Actually, user needs to load its model, optimizer, lr scheduler and the trainer as 
```python
@engine.on(Events.STARTED)
def resume_training(engine):
    engine.state.iteration = resume_epoch * len(engine.state.dataloader)
    engine.state.epoch = resume_epoch
```
No particular need to resume evaluator.

HTH

**Comment by vfdev-5** [comment]
I close the issue as answered. Feel free to reopen if needed more support on this
